src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java

```diff
@@ -530,16 +530,15 @@ private long nextIndexWithNextAvailableCycle0(final int cycle) {
             nextIndex = nextIndexWithinFoundCycle(nextCycle);
         else
             try {
-                int nextCycle0 = this.cycle;
-                while (true) {
```

Comment:
@nicktindall did this loop ever reach a second iteration or always breaks prior to that?

Reply:
The problem arose when this loop did make it to a second iteration, due to the first cycle being un-acquirable because of a bad header.

one such sequence of events was (two threads, cycle 3 is last cycle, with a bad header)

|tailer|appender|
|--------|------|
| | acquire cycle 3 for write|
|acquire cycle 3 for read|   |
|  | acquire 3 times out waiting for header|
|  | move invalid cycle 3 file|
|  | create/populate cycle 3 |
|  | create/populate cycle 4 |
|acquire 3 times out waiting for header | |
|assume 3 was deleted, skip to 4| | 

on that last row you just  skipped over the entries in (now valid) 3. It's an obscure sequence of events but can happen with very short (e.g. secondly) roll cycles, or in cases where the Appender is a replicator. Both of these things are true in UngracefulShutdownTest.

This change removes the assumption that `cycle(nextCycle0)` only returns false when the file doesn't exist.